library/std/src/io/util/tests.rs

```diff
@@ -124,6 +124,68 @@ fn empty_seeks() {
     assert!(matches!(e.seek(SeekFrom::Current(i64::MAX)), Ok(0)));
 }
 
+#[test]
+fn null_sinks() {
+    let mut n = null();
+    assert_eq!(n.write(&[]).unwrap(), 0);
+    assert_eq!(n.write(&[0]).unwrap(), 1);
+    assert_eq!(n.write(&[0; 1024]).unwrap(), 1024);
+    assert_eq!(Write::by_ref(&mut n).write(&[0; 1024]).unwrap(), 1024);
+}
+
+#[test]
+fn null_reads() {
+    let mut n = null();
+    assert_eq!(n.read(&mut []).unwrap(), 0);
+    assert_eq!(n.read(&mut [0]).unwrap(), 0);
+    assert_eq!(n.read(&mut [0; 1024]).unwrap(), 0);
+    assert_eq!(Read::by_ref(&mut n).read(&mut [0; 1024]).unwrap(), 0);
+
+    let mut buf = [];
+    let mut buf = ReadBuf::uninit(&mut buf);
+    n.read_buf(&mut buf).unwrap();
+    assert_eq!(buf.filled_len(), 0);
+    assert_eq!(buf.initialized_len(), 0);
+
+    let mut buf = [MaybeUninit::uninit()];
+    let mut buf = ReadBuf::uninit(&mut buf);
+    n.read_buf(&mut buf).unwrap();
+    assert_eq!(buf.filled_len(), 0);
+    assert_eq!(buf.initialized_len(), 0);
+
+    let mut buf = [MaybeUninit::uninit(); 1024];
+    let mut buf = ReadBuf::uninit(&mut buf);
+    n.read_buf(&mut buf).unwrap();
+    assert_eq!(buf.filled_len(), 0);
+    assert_eq!(buf.initialized_len(), 0);
+
+    let mut buf = [MaybeUninit::uninit(); 1024];
+    let mut buf = ReadBuf::uninit(&mut buf);
+    Read::by_ref(&mut n).read_buf(&mut buf).unwrap();
+    assert_eq!(buf.filled_len(), 0);
+    assert_eq!(buf.initialized_len(), 0);
+}
+
+#[test]
+fn null_seeks() {
+    let mut n = null();
+    assert!(matches!(n.seek(SeekFrom::Start(0)), Ok(0)));
+    assert!(matches!(n.seek(SeekFrom::Start(1)), Ok(0)));
+    assert!(matches!(n.seek(SeekFrom::Start(u64::MAX)), Ok(0)));
+
+    assert!(matches!(n.seek(SeekFrom::End(i64::MIN)), Ok(0)));
+    assert!(matches!(n.seek(SeekFrom::End(-1)), Ok(0)));
+    assert!(matches!(n.seek(SeekFrom::End(0)), Ok(0)));
+    assert!(matches!(n.seek(SeekFrom::End(1)), Ok(0)));
+    assert!(matches!(n.seek(SeekFrom::End(i64::MAX)), Ok(0)));
+
+    assert!(matches!(n.seek(SeekFrom::Current(i64::MIN)), Ok(0)));
+    assert!(matches!(n.seek(SeekFrom::Current(-1)), Ok(0)));
+    assert!(matches!(n.seek(SeekFrom::Current(0)), Ok(0)));
+    assert!(matches!(n.seek(SeekFrom::Current(1)), Ok(0)));
+    assert!(matches!(n.seek(SeekFrom::Current(i64::MAX)), Ok(0)));
```

Comment:
nit: to reduce duplication, maybe use a for loop?

Reply:
i just copied the test cases directly from the `io::Sink`/`io::Empty` ones.